### PR TITLE
update reticle from hand controllers only when triggered

### PR DESCRIPTION
**Pull request by howard-stearns**
Only update the (invisible) cursor position from hand controllers when the trigger is partially squeezed.

### CODE REVIEW

**Comment by howard-stearns** [comment]
This will make hud buttons and context menu not respond as you wave an untriggered hand controller around.

**Comment by howard-stearns** [comment]
test plan (in vive hmd)
- Open a QML window with buttons, such as the Running Scripts dialog. Wave a partially triggered hand controller (blue laser) over the buttons and notice that the buttons react to being "hovered".
- Release the buttons and wave over the same places, and notice that the buttons don't change their response to untriggered pointer movement.
- Drag hud windows around, interact with them, and make sure they operate normally.
- Press the context menu button (above trackpad on controller). It should come up where you are pointing. There should be no highlighting changes as you point the controller up and down. When you partially squeeze to get the blue laser, it should highlight the selectable submenu item.
- Interact with menus and make sure they operate normally.
- Repeat with the other hand.

File: scripts/system/controllers/handControllerPointer.js [hunk 1]
```diff
@@ -187,6 +187,32 @@ function overlayFromWorldPoint(point) {
...
+        return;
+    }
+    var hudPoint2d = overlayFromWorldPoint(hudPoint3d);
+
+    // We don't know yet if we'll want to make the cursor or laser visble, but we need to move it to see if
+    // it's pointing at a QML tool (aka system overlay).
+    setReticlePosition(hudPoint2d);
+    return hudPoint2d;
+}
+
```
Comment (howard-stearns) [inline scripts/system/controllers/handControllerPointer.js]: The above is all code that got abstracted out from update(), originally line 411 below.

File: scripts/system/controllers/handControllerPointer.js [hunk 2]
```diff
@@ -408,28 +449,13 @@ function update() {
     }
     leftTrigger.update();
     rightTrigger.update();
-    var controllerPose = Controller.getPoseValue(activeHand);
-    // Valid if any plugged-in hand controller is "on". (uncradled Hydra, green-lighted Vive...)
-    if (!controllerPose.valid) {
-        return off(); // Controller is cradled.
+    if (!activeTrigger.state) {
+        return off(); // No trigger
     }
```
Comment (howard-stearns) [inline scripts/system/controllers/handControllerPointer.js]: This is actually the major change (other than the context menu fun). Moving this up from original line 449 is what makes us not update the reticle position when there's not triggering.

**Comment by RyanDowne** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;7vPu6FcC&#x2F;61-can-we-do-anything-about-2d-items-reacting-to-reticle-position-when-not-triggered"> can we do anything about 2d items reacting to reticle position when not triggered?</a>

**Comment by hifi-gustavo** [comment]
Build console output for commit: 799686f29d46eb53c96742a53cea1bf9b4a7eea4 available at: https://gist.github.com/e48c7fb1421a18d2cc65312620741452<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8147.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8147.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5051f2f326eb7630a94c0973552b976322c39b74 available at: https://gist.github.com/e48c7fb1421a18d2cc65312620741452<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8147.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8147.dmg

**Comment by thoys** [comment]
# QA Testing report

- Successfully get the hover effect on QML buttons while squeezing the trigger slightly. :+1: 
- Releasing on a hovered button, keeps the button hovered (which means the cursor is not being updated when not squeezed) :+1: 
- Windows drag around normally :+1: 
- Context menu worked fine :+1: 
- Menu's work fine :+1:
- Both hands. :+1: 

- First run of the PR it kind of worked all sluggish. I had a browser window disappearing mysteriously which might have caused it. The second run it all worked as expected. Since I got no repro and no evidence that this PR caused it I will not give a fail for that.

QA Passed

**Comment by RyanDowne** [comment]
Only update the (invisible) cursor position from hand controllers when the trigger is partially squeezed.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5556bbb0a35f582fa7c42308137d3b18fed9c0d0 available at: https://gist.github.com/e48c7fb1421a18d2cc65312620741452<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8147.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8147.dmg

**Pull request merged**